_posts/2024-12-19-cosmicqc.md

```diff
@@ -0,0 +1,313 @@
+
+---
+
+## Introduction to coSMicQC
+
+After 2 and half years of being in my position and performing image-based profiling, I have come to realize that I am not perfect (shocking I know! 😱).
+No matter how much I try to manually optimize my segmentation parameters in CellProfiler or when using [Cellpose](https://www.cellpose.org/) or [Stardist](https://stardist.net/), it will not work for all of the single cells in my datasets.
+Now, I am not necessarily saying this is a fault of my own 😉, but segmentation errors are a persistent issue and it is currently a problem that hasn't been solved to this day.
+
+My point is, you can expect all of your datasets to contain some poorly segmented cells that can be described as technical artifacts.
+You can also expect that debris or smudges could also end up being mis-segmented as cells.
+When working with single-cell data, we do not want these technical artifacts impacting the interpretation of our results.
+Thus, was born this simple idea in my head.
+
+Why not just filter out single cells based on the extracted morphology features?
+
+You wouldn't need to train any model or perform any other pipeline.
+You already have the data in your lap to use, but how you use that data is what is important.
+
+In arriving to the stage, [coSMicQC](https://github.com/WayScience/coSMicQC), or Single cell Morphology Quality Control!
+
+{%
+  include figure.html
+  image="images/blog/cosmicqc/cosmicqc_stage.png"
+  width="100%"
+%}
+> Logo for coSMicQC being shown on a stage, like the star of the show! ⭐️
+
+I want to shout out Dave Bunten (aka @d33bs) here, who has been a major contributor to coSMicQC, which is an open-source software that we are developing with sustainable software development practices! 📣
+
+In summary, coSMicQC uses extracted morphology features directly from CellProfiler, performs normalization of specific features that are most related to detecting poor segmentation, and uses standard deviation to identify single cells that are considered technical outliers.
+
+Going into more detail on this new method, it works by first creating conditions.
+A condition is a feature or group of features that will capture a specific type of technical artifact.
+For example, nuclei area and intensity can work together as a condition to detect large nuclei with high intensity which represents segmentation of clustered/overlapping nuclei.
+Your condition will also need how many standard deviations from the mean that you expect these outliers to be located.
+If you want to detect cells above a mean, you use a positive value, and vice versa for cells below the mean.
+
+{%
+  include figure.html
+  image="images/blog/cosmicqc/area_outlier_example.png"
+  width="100%"
+%}
+> Example of how outliers are detected based on nuclei area in coSMicQC.
+
+You can use the function `find_outliers` to the simple conditions and voila, coSMicQC yields a dataframe with flagged single-cell outliers per condition.
```

Comment:
just checking that this is still the correct function name

```suggestion
You can use the function `find_outliers()` to the simple conditions and voila, coSMicQC yields a dataframe with flagged single-cell outliers per condition.
```

Reply:
It is correct! I will add that suggestion!